front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/pipeline/DefaultPipelineStrategyDAO.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.model.pipeline;
+
+import com.netflix.spinnaker.front50.exception.NotFoundException;
+import com.netflix.spinnaker.front50.model.ObjectType;
+import com.netflix.spinnaker.front50.model.StorageService;
+import com.netflix.spinnaker.front50.model.StorageServiceSupport;
+import rx.Scheduler;
+
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class DefaultPipelineStrategyDAO extends StorageServiceSupport<Pipeline> implements PipelineStrategyDAO {
+  public DefaultPipelineStrategyDAO(StorageService service,
+                            Scheduler scheduler,
+                            int refreshIntervalMs) {
+    super(ObjectType.STRATEGY, service, scheduler, refreshIntervalMs);
+  }
+
+  @Override
+  public String getPipelineId(String application, String pipelineName) {
+    Pipeline matched = getPipelinesByApplication(application)
+      .stream()
+      .filter(pipeline -> pipeline.getName().equalsIgnoreCase(pipelineName))
+      .findFirst()
+      .orElseThrow(() -> new NotFoundException(
+        String.format("No pipeline strategy found with name '%s' in application '%s'", pipelineName, application)
+      ));
+
+    return matched.getId();
+  }
+
+  @Override
+  public Collection<Pipeline> getPipelinesByApplication(String application) {
+    return all()
+      .stream()
+      .filter(pipelineStrategy -> pipelineStrategy.getApplication().equalsIgnoreCase(application))
+      .collect(Collectors.toList());
+  }
+
+  @Override
+  public Collection<Pipeline> getPipelineHistory(String id, int limit) {
```

Comment:
here you call it getPIpelineHistory
in snapshot below you are just getHistory
in notification there is no history.

When is there and is there not history?
At least in GCS, history is a bucket level so in fact everythig nwill have a history whether or not we access it.
Why not just add getHistory to the base StorageServiceSupport? These methods just pass through anyway. You can override when you cannot support history for whatever reason, or need to do so differently.